week2_and_week3/Singleton/singleton.py

```diff
@@ -0,0 +1,31 @@
+from typing import Dict
+
+class Singleton:
+    """The singleton class"""
+    
+    _shared_data: Dict[str, str] = {}
```

Comment:
to jest po prostu zmienna statyczna (klasowa), czyli wspólna zmienna klasy dzielona przez instancje